### PR TITLE
refactor(influxdb): Refactor trace code for clarity and reliability

### PR DESCRIPTION
**Pull request by ayang64**
Refactor trace code for clarity and reliability.  The branch is misnamed unfortunately.

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

File: cmd/influxd/run/server.go [hunk 1]
```diff
@@ -581,47 +585,56 @@ type Service interface {
...
-		log.Printf("writing CPU profile to: %s\n", cpuprofile)
-		prof.cpu = f
-		pprof.StartCPUProfile(prof.cpu)
+
+		s.CPUProfileFile = f
+		if err := pprof.StartCPUProfile(s.CPUProfileFile); err != nil {
+			return err
+		}
+
+		log.Printf("writing CPU profile to: %s\n", s.CPUProfile)
```
Comment (e-dard) [inline cmd/influxd/run/server.go]: Can you tidy this up? `Writing...`

File: cmd/influxd/run/server.go [hunk 2]
```diff
@@ -581,47 +585,56 @@ type Service interface {
...
-			log.Fatalf("memprofile: %v", err)
+			return fmt.Errorf("memprofile: %v", err)
 		}
-		log.Printf("writing mem profile to: %s\n", memprofile)
-		prof.mem = f
+
+		s.MemProfileFile = f
 		runtime.MemProfileRate = 4096
+
+		log.Printf("writing mem profile to: %s\n", s.MemProfile)
```
Comment (e-dard) [inline cmd/influxd/run/server.go]: Ditto on capitalisation here.

File: cmd/influxd/run/server.go [hunk 3]
```diff
@@ -89,8 +89,10 @@ type Server struct {
 	reportingDisabled bool
 
 	// Profiling
-	CPUProfile string
-	MemProfile string
+	CPUProfile     string
+	CPUProfileFile *os.File
```
[the remark this answers is not in the document]
Comment (ayang64) [reply cmd/influxd/run/server.go]: agreed.

File: cmd/influxd/run/server.go (hunk 2 above)
Comment (ayang64) [reply cmd/influxd/run/server.go]: yup

File: cmd/influxd/run/server.go (hunk 1 above)
Comment (ayang64) [reply cmd/influxd/run/server.go]: yup.

**Pull request merged**